src/hotspot/share/gc/shared/stringdedup/stringDedupProcessor.hpp

```diff
@@ -51,7 +53,7 @@ class StringDedup::Processor : public CHeapObj<mtGC> {
   NONCOPYABLE(Processor);
 
   static OopStorage* _storages[2];
-  static StorageUse* volatile _storage_for_requests;
+  static DeferredStatic<Atomic<StorageUse*>> _storage_for_requests;
```

Comment:
What is the reason for the `DeferredStatic`?

Is it just for the init once semantics and asserts?

_Is this a style we should adapt more broadly for our init once static variables?_

Reply:
We discussed this a bit more, and decided to make the Atomic ctors constexpr
so that, with constant arguments, they are guaranteed to be static initialized.
And then change this to remove the use of DeferredStatic and instead have the
uses assert non-null the way they used to, instead of using the DeferredStatic
initialization check. The benefit is that many uses of non-local Atomic
definitely don't need the additional DeferredStatic ceremony.